src/output-json-alert.c

```diff
@@ -173,6 +174,25 @@ static void AlertJsonSourceTarget(const Packet *p, const PacketAlert *pa,
     jb_close(js);
 }
 
+static void AlertJsonReference(
+        AlertJsonOutputCtx *json_output_ctx, const PacketAlert *pa, JsonBuilder *jb)
+{
+    if (!pa->s->references) {
+        return;
+    }
+
+    const DetectReference *kv = pa->s->references;
+    jb_open_array(jb, "references");
+    while (kv) {
+        const size_t size = strlen(kv->key) + strlen(kv->reference) + 1;
+        char kv_store[size];
+        snprintf(kv_store, size, "%s%s", kv->key, kv->reference);
```

Comment:
Should there not be a separator like `:` between key and value ?

Reply:
No ... An example key is `https://` and the corresponding value might be `github.com/OISF/suricata`

Reply:
So, key is rather a protocol scheme ?

Reply:
Isn't the scheme more like `url`, `md5`, `cve`?